src/parser/DASHTree.cpp

```diff
@@ -1691,6 +1691,11 @@ void adaptive::CDashTree::RefreshLiveSegments()
                   LOG::LogF(LOGDEBUG,
                             "MPD update - No new segments (repr. id \"%s\", period id \"%s\")",
                             repr->GetId().data(), period->GetId().data());
+                  if (repr->IsWaitForSegment() &&
+                      period->GetId() != m_periods.back()->GetId())
+                  {
+                    repr->SetIsWaitForSegment(false);
+                  }
```

Comment:
you havent provide any explanation nor manifests that explain why this is needed
when there are no segments you must not remove the "wait for segments"
the demxumer callbacks must be kept stopped